MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,455 @@
+
+                        Abbreviations =
+                            m.TypeParameterCount == 0
+                            ? string.Empty
+                            : " where " + string.Join(", ", from a in m.TypeParameterAbbreviationByName
+                                                            select a.Value + " = " + a.Key),
+                    }
+                    into e
+                    select e.SourceOrder + ": "
+                         + e.Name + e.TypeParameters + e.Parameters + e.Abbreviations)
+                {
+                    Console.Error.WriteLine(m);
+                }
+            }
+
+            var indent = new string('\x20', 4);
+            var indent2 = indent + indent;
+            var indent3 = indent2 + indent;
+
+            var baseImports = new []
+            {
+                "System",
+                "System.CodeDom.Compiler",
+                "System.Collections.Generic",
+            };
+
+            var imports =
+                from ns in baseImports.Concat(usings)
+                select indent + $"using {ns};";
+
+            var classes =
+                from md in q
+                select md.Method.Syntax into md
+                group md by (string) md.Identifier.Value into g
+                select new
+                {
+                    Name = g.Key,
+                    Overloads =
+                        from md in g
+                        select
+                            MethodDeclaration(md.ReturnType, md.Identifier)
+                                .WithAttributeLists(md.AttributeLists)
+                                .WithModifiers(md.Modifiers)
+                                .WithTypeParameterList(md.TypeParameterList)
+                                .WithConstraintClauses(md.ConstraintClauses)
+                                .WithParameterList(md.ParameterList)
+                                .WithExpressionBody(
+                                    ArrowExpressionClause(
+                                        InvocationExpression(
+                                            MemberAccessExpression(
+                                                SyntaxKind.SimpleMemberAccessExpression,
+                                                IdentifierName("MoreEnumerable"),
+                                                IdentifierName(md.Identifier)),
+                                            ArgumentList(SeparatedList(
+                                                from p in md.ParameterList.Parameters
+                                                select Argument(IdentifierName(p.Identifier)))))
+                                            .WithLeadingTrivia(Space))
+                                        .WithLeadingTrivia(Whitespace(indent3)))
+                                .WithSemicolonToken(ParseToken(";").WithTrailingTrivia(LineFeed))
+                }
+                into m
+                select (!noClassLead ? $@"
```

Comment:
Double negatives are hard to parse. Given that this is the only use, it could be renamed to `classLead` to make this statement clearer.

Reply:
All Boolean options are initialised to `false`, thus the reason for an inverted name like `noClassLead`. I agree though, double-negatives are hard to read. Would you introduce `var classLead = !noClassLead`?

Reply:
I would replace the initialization to `false` with an initialization to `true`.